src/engine.c

```diff
@@ -34,13 +34,48 @@ char *exp_names[MAX_EXPANSION + 1] =
 	"The Brink of War",
 	"Alien Artifacts",
 	"Xeno Invasion",
+	"Rebel vs Imperium (without Gathering Storm)",
 	NULL
 };
 
+int exp_display_order[MAX_EXPANSION] = {0, 1, 2, 6, 3, 4, 5};
+
 /*
  * Maximum number of players per expansion
  */
-int exp_max_player[MAX_EXPANSION] = {4, 5, 6, 6, 5, 5};
+int exp_max_player[MAX_EXPANSION] = {4, 5, 6, 6, 5, 5, 5};
+
+/*
+ * Check whether the expansion can be played with goals.
+ */ 
+inline int expansion_has_goals(int exp)
```

Comment:
Great refactoring! But I would rather see these settings in one struct per expansion, to group related information together. expansion_has_goals(exp) should then be replaced with exp_data[exp].has_goals (for instance).

(Using designated initializers, fields set to FALSE can be omitted.)

Information needed:
 * name
 * short_name
 * max_players
 * has_goals
 * has_takeovers
 * has_prestige
 * has_start_world_choice
 * prev_in_arc (perhaps not needed, it would turn the logic in initialization upside down)